homework4-1.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"fmt"
+	"math"
+)
+
+type Figure interface {
+	area() float64
+	perimeter() float64
+}
+
+type Square struct {
+	a float64
+}
+
+func (s Square) area() float64 {
+	return s.a*s.a
+}
+
+func (s Square) perimeter() float64 {
+	return 4*s.a
+}
+
+type Circle struct {
+	radius float64
+}
+
+func (c Circle) area() float64 {
+	return math.Pi*c.radius*c.radius
+}
+
+func (c Circle) perimeter() float64 {
+	return 2*c.radius*math.Pi
+}
+
+type Rectangle struct {
+	a float64
+	b float64
+}
+
+func (r Rectangle) perimeter() float64 {
+	return 2*(r.a+r.b)
+}
+
+func (r Rectangle) area() float64 {
+	return r.a*r.b
+}
+
+
+func main() {
+
+	var s Figure = Square{1}
+	var c Figure = Circle{1}
+	var r Figure = Rectangle{3,4}
```

Comment:
This case is pain. Generally, it's bad practice to create struct objects with unnamed params.
It shouldn't be fixed due to task description, just keep in mind